src/Cache/FileCacheManager.php

```diff
@@ -70,7 +70,7 @@ public function __construct(
         $this->handler = $handler;
         $this->signature = $signature;
         $this->isDryRun = $isDryRun;
-        $this->cacheDirectory = $cacheDirectory ?: new Directory(dirname(realpath($handler->getFile())));
+        $this->cacheDirectory = $cacheDirectory ?: new Directory(dirname($handler->getFile()));
```

Comment:
@keradus 

The problem is that the cache file may not exist, and then `realpath()` returns `false`, meaning that when a cache hasn't been created yet, paths aren't resolved to relative to the location of the cache file.